runtime/lua/vim/lsp/log.lua

```diff
@@ -31,21 +31,37 @@ do
     return logfilename
   end
 
-  vim.fn.mkdir(vim.fn.stdpath('cache'), "p")
-  local logfile = assert(io.open(logfilename, "a+"))
-
-  local log_info = vim.loop.fs_stat(logfilename)
-  if log_info and log_info.size > 1e9 then
-    local warn_msg = string.format(
-      "LSP client log is large (%d MB): %s",
-      log_info.size / (1000 * 1000),
-      logfilename
-    )
-    vim.notify(warn_msg)
+  local logfile, openerr
+  ---@private
+  --- Opens log file. Returns true if file is open, false on error
+  local function open_logfile()
+    -- Try to open file only once
+    if logfile then return true end
+    if openerr then return false end
+
+    vim.fn.mkdir(vim.fn.stdpath('cache'), "p")
```

Comment:
```
Error executing luv callback:
...hi/data/github/neovim/neovim/runtime/lua/vim/lsp/log.lua:42: E5560: vimL function must not be calle
d in a lua loop callback
```

Reply:
We'd have to use libuv's `fs_mkdir`, but I believe it's not recursive, so we'd maybe need a new utility function that does this.

Maybe for now it would be good enough to just move it out of this function and create the directory when initializing `vim.lsp.log` module?